src/main/java/repo/impl/PostgresSessionRepo.java

```
@@ -0,0 +1,129 @@
+    }
+
+    @Override
+    public List<SerializableFormSession> findUserSessions(String username) {
+        List<SerializableFormSession> sessions = this.jdbcTemplate.query(
+                replaceTableName("SELECT * FROM %s WHERE username = ?"),
+                new Object[] {username},
+                new SessionMapper());
+        return sessions;
+    }
+
+    @Override
+    public void delete(String id) {
+        this.jdbcTemplate.update(replaceTableName("DELETE FROM %s WHERE id = ?"), id);
+    }
+
+    // helper class for mapping a db row to a serialized session
+    private static final class SessionMapper implements RowMapper<SerializableFormSession> {
+
+        public SerializableFormSession mapRow(ResultSet rs, int rowNum) throws SQLException {
+
+            SerializableFormSession session = new SerializableFormSession();
+            session.setId(rs.getString("id"));
+            session.setInstanceXml(rs.getString("instanceXml"));
+            session.setFormXml(rs.getString("formXml"));
+            session.setRestoreXml(rs.getString("restoreXml"));
+            session.setUsername(rs.getString("username"));
+            session.setInitLang(rs.getString("initLang"));
+            session.setSequenceId(Integer.parseInt(rs.getString("sequenceId")));
+            session.setDomain(rs.getString("domain"));
+            session.setPostUrl(rs.getString("postUrl"));
+
+            byte[] st = (byte[]) rs.getObject("sessionData");
+
+            if(st != null) {
+                ByteArrayInputStream byteInputStream = new ByteArrayInputStream(st);
+                ObjectInputStream objectInputStream;
+                try {
+                    objectInputStream = new ObjectInputStream(byteInputStream);
+                    Map<String, String> sessionData = (HashMap) objectInputStream.readObject();
+                    session.setSessionData(sessionData);
+                } catch (IOException e) {
```

going to bubble these up - should fail hard on this since loading the form later without session data will probably crash